Content.Shared/Preferences/HumanoidCharacterProfile.cs

```diff
@@ -534,9 +534,16 @@ public void EnsureValid(ICommonSession session, IDependencyCollection collection
                 name = GetName(Species, gender);
             }
 
-            var customspeciename = speciesPrototype.CustomName
-                ? FormattedMessage.RemoveMarkup(Customspeciename ?? "")[..MaxNameLength]
-                : "";
+            string customspeciename;
+            if (string.IsNullOrEmpty(Customspeciename))
+                customspeciename = "";
+            else if (Customspeciename.Length > MaxNameLength)
+                customspeciename = FormattedMessage.RemoveMarkup(Customspeciename)[..MaxNameLength];
+            else
+                customspeciename = FormattedMessage.RemoveMarkup(Customspeciename);
+
+            if (!speciesPrototype.CustomName)
+                customspeciename = "";
```

Comment:
For the love of god, just do
```suggestion
            var customspeciename = speciesPrototype.CustomName ? FormattedMessage.RemoveMarkup(Customspeciename ?? "") : "";
                
            if (Customspeciename.Length > MaxNameLength)
                customspeciename = customspeciename[..MaxNameLength];
```

Reply:
This resulted in return null and crashed.

Reply:
```suggestion
            var customspeciename =
                !speciesPrototype.CustomName
                || string.IsNullOrEmpty(Customspeciename)
                    ? ""
                    : Customspeciename.Length > MaxNameLength
                        ? FormattedMessage.RemoveMarkup(Customspeciename)[..MaxNameLength]
                        : FormattedMessage.RemoveMarkup(Customspeciename);
```